### PR TITLE
Extend IBAN support

### PR DESCRIPTION
**Pull request by Corniel**
Added specific support for Russia, Sudan, and the Central African Republic. Because Russia's IBAN has a length of 33, it was rejected by the generic validation assuming that the maximum length was 32. This has been changed to 36.

Burundi's pattern turned out to be wrong.

### CODE REVIEW

File: IBAN.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+﻿# International Bank Account Number
+An overview of all IBAN's supported. Note that for countries not mentioned, IBA
```
Comment (LauraXiulan) [inline IBAN.md:2]: Missing text? The sentence seems to stop abruptly

File: IBAN.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ISO 7064 MOD-97-10
+
+# Basic Bank Account Number
+Abbreviated as BBAN, is a way of describing the IBAN (for a specific country).
+* a: alphabetic
+* c: alphanumeric 
+* n: numeric
+
+In this table, for countries with fixed characters (such as Costa Rica) those
+are displayed as `[?]`. This is strictly not BBAN, but helps when genrating
```
Comment (LauraXiulan) [inline IBAN.md:13]: generating

File: src/Qowaiv/Financial/InternationalBankAccountNumber.Patterns.cs [hunk 3]
```diff
@@ -19,16 +19,17 @@ namespace Qowaiv.Financial;
         Bban(Country.AL, "8n,16c"),
         Bban(Country.AO, "21n"),
         Bban(Country.AT, "16n"),
-        Bban(Country.AZ, "4c,20n"),
+        Bban(Country.AZ, "4a,20c"),
```
Comment (LauraXiulan) [inline src/Qowaiv/Financial/InternationalBankAccountNumber.Patterns.cs:22]: Why so many changes in patterns, the description of this PR only mentions four countries?

**Review by LauraXiulan — CHANGES_REQUESTED** [review]
See other comments

File: src/Qowaiv/Financial/InternationalBankAccountNumber.Patterns.cs (hunk 3 above)
Comment (Corniel) [reply src/Qowaiv/Financial/InternationalBankAccountNumber.Patterns.cs:22]: I did not want to mention this kind of (relatively) small changes. From numeric to alphanumeric obviously ment that previously IBAN's could have been rejected incorrectly (the other way around is not an issue). But you've got a point, these are also changes.

File: IBAN.md (hunk 1 above)
Comment (Corniel) [reply IBAN.md]: Is this better?

Comment (LauraXiulan) [reply IBAN.md]: Yes

**Review by LauraXiulan — APPROVED** [review]

**Pull request merged**